GitUpdateScript.sh

```diff
@@ -1,122 +1,108 @@
-		then
-		 git stash # saves changes but does not commit before changing branch
-		 git checkout $branch_I_want
-		elif [[ "$flagWhatBranchOn" -eq "$FlagOnBranchDev" ]]
-		then
-		 git stash # saves changes but does not commit before changing branch
-		 git checkout $branch_I_want_dev
-		fi
-	fi
-}
 
+    local flagRequiredBranchFound=0
+
+    if [[ ${1} -gt 0 ]]
+    then
+        echo "Requested branch found: "${2}
+        CheckoutBranchAndPull ${2}
+        local flagRequiredBranchFound=1
+    else
+        echo "Moving to Default Branch"
+        MoveToDefaultBranches
+    fi
+
+    if [[ ${flagRequiredBranchFound} -eq 0 ]]
+    then
+        echo "WARNING NO BRANCHES FOUND"
+    fi
+}
 
-function BranchChecking(){
-	#constants
-	flagDontKnowBranch=0
-	flagOnCorrectBranch=1
-	flagOnBranchMaster=2
-	FlagOnBranchDev=3
 
+function DoesBranchExistForParameterPassedIntoScript(){
+    #if used to check if a parameter is passed into script
+    #this is used to prevent error
+    if [[ ${#1} -eq 0 ]]  #${#var}  Use the length of var.
+    then
+        local branchExist=0
+    else
+        local branchExist=$(git branch -a | grep ${1} | wc -l)
+    fi
 
-	flagWhatBranchOn="$flagDontKnowBranch"
-	echo "Current Branches"
-	git branch| tee log_branch
-	branch_I_want="master"
-	branch_I_want_dev="dev"
+    echo $branchExist #acting as a return
+}
 
+function GoToGitRepositoryDirectory(){
 
-	while read -r branch_option
-	do # will make sure we are on the master branch and change flag
+	cd $1${2:1:-4} #change to git dir
+	echo "Starting git repository refresh @" $(pwd) #print dir to screen
+}
 
-	FindsWhatBranchRepositoryIsOn
+function PrintParameterPassedIntoScript(){
+    if [[ ${#1} -gt 0 ]]  #${#var}  Use the length of var.
+    then
+        echo "Will try moving to branch: ${1}"
+    fi
+}
 
-	done < log_branch  #end of loop
-	rm log_branch #rm removes/dels the file
 
-	MoveToCorrectBranch
 
-}
-function FindGitReposotories()
-{
-    arrayFileDir=($(find -name .git))
-    numberOfGitRepos=$(find -name .git | wc -l)  #wc is word count -l makes it count lines, | is pipe and find finds all .git
+#########################################
+##########   Script Start  ##############
+#########################################
 
-    echo "Going to each git repository and updating:"
+homeDir=$(pwd)  #v1 holds the home dir
 
-    for (( n=0; n<$numberOfGitRepos; n++ ))
-    do
-        file_dir=${arrayFileDir[$n]}
-        GoToGitRepositoryDirectory
+requestedBranchName=$1
 
-        BranchChecking  #call branch checking function
+arrayGitDir=($(find -name .git))
+numberOfGitRepos=$(find -name .git | wc -l)  #wc is word count -l makes it count lines, | is pipe and find finds all .git
 
-        git pull
+echo "Going to each git repository and updating:"
 
-        echo "Elapsed Time:" $SECONDS
-        echo ""
-    done
-}
+PrintParameterPassedIntoScript ${requestedBranchName}
 
+for (( n=0; n<$numberOfGitRepos; n++ ))
+do
+    gitDir=${arrayGitDir[$n]}
```

Comment:
positional parameters and returns are now being used so the code looks very different to the master branch, but it's really not that different.